controllers/sensor/resource.go

```diff
@@ -141,14 +143,17 @@ func (ctx *sensorContext) makeLegacyDeploymentSpec() appv1.DeploymentSpec {
 
 // deploymentBuilder builds the deployment specification for the sensor
 func (ctx *sensorContext) deploymentBuilder() (*appv1.Deployment, error) {
-	deploymentSpec := ctx.makeDeploymentSpec()
+	deploymentSpec, err := ctx.makeDeploymentSpec()
+	if err != nil {
+		return nil, err
+	}
 	deployment := &appv1.Deployment{
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace:    ctx.sensor.Namespace,
-			GenerateName: fmt.Sprintf("%s-sensor-", ctx.sensor.Name),
+			GenerateName: fmt.Sprintf("%s-", ctx.sensor.Name),
```

Comment:
I don't have a strong opinion about either Gateway or Sensor deployments' `GenerateName`, and unlike Service, giving both Gateway and Sensor a same name won't cause any issue (some of the users might do that), but it will be harder for the users to identify which one is for Gateway, and which one is for Sensor.